File: Java/Problem001-010/Problem003/Main.java

public class Main {

	public static void main(String[] args) {
		final long NUM = 600851475143L;
		long largestPrimeFactor = -1;
		
		for(long i = 1; i < Math.sqrt(NUM); i++) {
			if(NUM % i == 0) {
				if(isPrime(i)) {
					largestPrimeFactor = i;
				}
			}
		}
		
		System.out.println("Largest Prime Factor: " + largestPrimeFactor);
	}
	
	public static boolean isPrime(long n) {
		if(n == 2 || n == 3 || n == 5 || n == 7) {
			return true;
		}
		else if (n % 2 == 0 || n == 1) {
			return false;
		}
		else {
			for(int i = 3; i <= Math.sqrt(n); i += 2) {
				if(n % i == 0) {
					return false;
				}
			}
			return true;
		}
	}

}

File: Python/Problem001-010/Problem003/Main.py
import constant
import math

largestPrimeFactor = -1

def isPrime(n):
	if n == 2 or n == 3 or n == 5 or n == 7:
		return True
	elif n % 2 == 0 or n == 1:
		return False
	else:
		for i in range(3,int(math.sqrt(n))):
			if n % i == 0:
				return False
		return True

for i in range(1,int(math.sqrt(constant.NUM))):
	if constant.NUM % i == 0:
		if isPrime(i):
			largestPrimeFactor = i

print(largestPrimeFactor)